modules/stochastic_tools/test/include/postprocessors/OptimizationTestFunction.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#pragma once
+
+#include "GeneralPostprocessor.h"
+
+class OptimizationTestFunction : public GeneralPostprocessor
```

Comment:
I know a lot of people are interested in these functions (Dewen was, Som as well, etc). 
I think Som actually tried to add the Rosenbrock function at some point. 
Do you think we could split this function into two different functions and move them to STM from test?
Actually I am fine having it in the test but I still think we should split this in 2 different functions instead of doing the if-else in here.

Reply:
Made two different function classes with this class being the base.